src/java.base/share/classes/javax/crypto/spec/RC2ParameterSpec.java

```diff
@@ -98,8 +98,13 @@ public RC2ParameterSpec(int effectiveKeyBits, byte[] iv) {
      */
     public RC2ParameterSpec(int effectiveKeyBits, byte[] iv, int offset) {
         this.effectiveKeyBits = effectiveKeyBits;
-        if (iv == null) throw new IllegalArgumentException("IV missing");
+        if (iv == null) {
+            throw new IllegalArgumentException("IV missing");
+        }
         int blockSize = 8;
+        if (offset < 0) {
+            throw new ArrayIndexOutOfBoundsException("offset is negative");
+        }
```

Comment:
Move these lines before line 104 as `blocksize` doesn't need to be set until after this check.

Reply:
This is a side issue, but it looks like this API can also throw `IndexOutOfBoundsException` if an offset is input which causes `System.arraycopy` (on line 112) to access the iv out of range. Please check and file a separate issue to have this exception documented, if so.

Reply:
I don't think this can happen because of the check on line 108.

Reply:
Ah yes, you're right.